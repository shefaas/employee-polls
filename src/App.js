import { useEffect, Fragment } from "react";
// import LoadingBar from "react-redux-loading-bar";
import { Routes, Route } from "react-router-dom";

import NavBar from "./components/NavBar";

function App() {
  return (
    <Fragment>
      {/* <LoadingBar /> */}
      <div className="container">
        <NavBar />
        <Routes>
          <Route exact path="/" element={<div>hola</div>} />
          <Route path="/questions/:question_id" element={<div></div>} />
          <Route path="/add" element={<div></div>} />
        </Routes>
      </div>
    </Fragment>
  );
}

export default App;
