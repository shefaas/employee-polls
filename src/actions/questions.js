import { showLoading, hideLoading } from "react-redux-loading-bar";
import { saveQuestionAnswer } from "../utils/api";

export const GET_QUESTIONS = "GET_QUESTIONS";
export const VOTE_ON_QUESTION = "VOTE_ON_QUESTION";

export function getQuestions(questions) {
  return {
    type: GET_QUESTIONS,
    questions,
  };
}

export function voteOnQuestion({ authedUser, qid, answer }) {
  return {
    type: VOTE_ON_QUESTION,
    voteData: {
      authedUser,
      qid,
      answer,
    },
  };
}
