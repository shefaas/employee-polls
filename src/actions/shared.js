import {
  getInitialData,
  saveQuestionAnswer,
  createQuestion,
} from "../utils/api";
import { getUsers, saveQuestionForUser, voteOnQuestionForUser } from "./users";
import { getQuestions, voteOnQuestion, saveQuestion } from "./questions";
import { setAuthedUser } from "./authedUser";
import { showLoading, hideLoading } from "react-redux-loading-bar";

export const VOTE_ON_QUESTION = "VOTE_ON_QUESTION";

// const AUTHED_USER_ID = "tylermcginnis";

export function handleInitialData() {
  return (dispatch) => {
    dispatch(showLoading());
    return getInitialData()
      .then(({ users, questions }) => {
        dispatch(getUsers(users));
        dispatch(getQuestions(questions));
        dispatch(hideLoading());
      })
      .catch((error) => {
        console.log({ error });
      });
  };
}

export function handleVoteOnQuestion(qid, answer) {
  return (dispatch, getState) => {
    const { authedUser } = getState();

    dispatch(showLoading());
    dispatch(voteOnQuestion({ authedUser, qid, answer }));
    dispatch(voteOnQuestionForUser({ authedUser, qid, answer }));

    return saveQuestionAnswer({ authedUser, qid, answer })
      .then(() => {
        dispatch(hideLoading());
      })
      .catch((error) => {
        console.log({ error });
      });
  };
}

export function handleSaveQuestion(option1, option2) {
  return (dispatch, getState) => {
    const { authedUser } = getState();
    dispatch(showLoading());

    return createQuestion({
      author: authedUser,
      optionOneText: option1,
      optionTwoText: option2,
    })
      .then((question) => {
        dispatch(saveQuestion(question));
        dispatch(saveQuestionForUser(question));
        dispatch(hideLoading());
      })
      .catch((error) => {
        console.log({ error });
      });
  };
}
