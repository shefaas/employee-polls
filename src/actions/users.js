import { VOTE_ON_QUESTION } from "./shared";

export const GET_USERS = "GET_USERS";
export const UPDATE_USER_QUESTION = "UPDATE_USER_QUESTION";

export function getUsers(users) {
  return {
    type: GET_USERS,
    users,
  };
}

export function voteOnQuestionForUser({ authedUser, qid, answer }) {
  return {
    type: VOTE_ON_QUESTION,
    voteData: {
      authedUser,
      qid,
      answer,
    },
  };
}

export function saveQuestionForUser(question) {
  return {
    type: UPDATE_USER_QUESTION,
    question,
  };
}
