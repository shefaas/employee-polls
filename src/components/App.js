import { useEffect, Fragment } from "react";
import LoadingBar from "react-redux-loading-bar";
import { Routes, Route } from "react-router-dom";
import { connect } from "react-redux";

import NavBar from "./NavBar";
import Home from "./Home";
import { handleInitialData } from "../actions/shared";

function App(props) {
  useEffect(() => {
    props.dispatch(handleInitialData());
    const { users, questions, authedUser } = props;
    console.log({ users });
    console.log({ questions });
    console.log({ authedUser });
  }, []);

  return (
    <Fragment>
      <LoadingBar />
      {props.loading === true ? null : (
        <div className="container">
          <NavBar />
          <Routes>
            <Route exact path="/" element={<Home />} />
            <Route path="/questions/:question_id" element={<div></div>} />
            <Route path="/add" element={<div></div>} />
          </Routes>
        </div>
      )}
    </Fragment>
  );
}

const mapStateToProps = ({ users, questions, authedUser }) => ({
  users,
  questions,
  authedUser,
  loading: authedUser === null,
});
export default connect(mapStateToProps)(App);
