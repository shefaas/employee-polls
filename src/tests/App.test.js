import { render, screen, fireEvent, getByTestId } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import "@testing-library/jest-dom";
import { createStore } from "redux";
import { Provider } from "react-redux";
import reducer from "../reducers";
import thunk from "redux-thunk";
import { applyMiddleware } from "redux";
import App from "../components/App";
import { setAuthedUser } from "../actions/authedUser";
import { handleInitialData } from "../actions/shared";

const store = createStore(reducer, applyMiddleware(thunk));

describe("Main tests from App", () => {
  test("Render App component for unauthorized users (redirect to login)", () => {
    render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );

    const title = screen.getByText("Employee Polls App ðŸ‘‹");

    expect(title).toBeInTheDocument();

    fireEvent.click(title);
    expect(screen.getByTestId("login-dropdown")).toBeInTheDocument();
  });

  test("Fire event to see login dropdown menu", () => {
    render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );

    const title = screen.getByText("Employee Polls App ðŸ‘‹");
    fireEvent.click(title);
    expect(screen.getByTestId("login-dropdown")).toBeInTheDocument();
  });

  it("Matches the snapshot when a parameter is passed", () => {
    expect(
      render(
        <Provider store={store}>
          <Router>
            <App testingValue={"Hello"} />
          </Router>
        </Provider>
      )
    ).toMatchSnapshot();
  });
});
