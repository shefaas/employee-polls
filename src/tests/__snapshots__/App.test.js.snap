// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main tests from App Matches the snapshot when a parameter is passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <p>
          Hello
        </p>
        <div
          style="display: flex; flex-direction: column; justify-content: center; align-items: center; align-content: center; width: 170vh; height: 80vh; row-gap: 50px;"
        >
          <div>
            <h2>
              Employee Polls App ðŸ‘‹
            </h2>
          </div>
          <div
            class="dropdown"
            data-testid="login-dropdown"
          >
            <button
              aria-expanded="false"
              class="dropdown-toggle btn btn-primary"
              id="dropdown-basic"
              type="button"
            >
              Select Login User
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <p>
        Hello
      </p>
      <div
        style="display: flex; flex-direction: column; justify-content: center; align-items: center; align-content: center; width: 170vh; height: 80vh; row-gap: 50px;"
      >
        <div>
          <h2>
            Employee Polls App ðŸ‘‹
          </h2>
        </div>
        <div
          class="dropdown"
          data-testid="login-dropdown"
        >
          <button
            aria-expanded="false"
            class="dropdown-toggle btn btn-primary"
            id="dropdown-basic"
            type="button"
          >
            Select Login User
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Main tests from App matches the snapshot when a name is passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <p>
          Hello
        </p>
        <div
          style="display: flex; flex-direction: column; justify-content: center; align-items: center; align-content: center; width: 170vh; height: 80vh; row-gap: 50px;"
        >
          <div>
            <h2>
              Employee Polls App ðŸ‘‹
            </h2>
          </div>
          <div
            class="dropdown"
            data-testid="login-dropdown"
          >
            <button
              aria-expanded="false"
              class="dropdown-toggle btn btn-primary"
              id="dropdown-basic"
              type="button"
            >
              Select Login User
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <p>
        Hello
      </p>
      <div
        style="display: flex; flex-direction: column; justify-content: center; align-items: center; align-content: center; width: 170vh; height: 80vh; row-gap: 50px;"
      >
        <div>
          <h2>
            Employee Polls App ðŸ‘‹
          </h2>
        </div>
        <div
          class="dropdown"
          data-testid="login-dropdown"
        >
          <button
            aria-expanded="false"
            class="dropdown-toggle btn btn-primary"
            id="dropdown-basic"
            type="button"
          >
            Select Login User
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
